content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,174 @@
+---
+weight: 60
+category: "extending-chat-widget"
+subcategory: "custom-identity-provider"
+title: "Custom Identity Provider"
+tagline: "Pair your own customer identity with LiveChat"
+desc: "Cross-platform chats for your applications"
+versionGroup: "chat-widget"
+---
+
+# Introduction
+
+## What is it for?
+
+If you have a more advanced service, handling user sessions, you can pair LiveChat customer identity with your own. As a result, your customers will never lose chats they made with your support. What is more, if you also use LiveChat on other platfroms (e.g., in mobile applications), their conversations will sync, making the communication even more seamless and natural.
+
+## Setup
+Customer Identity Provider is essentially a singleton function. You can define as a part of `LiveChatWidget` object from the tracking code used to install ChatWidget on your website or any other place accessing `window.__lc` object, however - **it has to be provided before widget initialization**.
+
+```js
+    // Custom Identity Provider
+    window.__lc.custom_identity_provider = function() {
+      return {
+          getToken: ...,
+          getFreshToken: ...,
+          hasToken: ...,
+          invalidate: ...
+      }
+    }
+```
+
+# Methods
+
+The aforementioned function assigned to `custom_identity_provider` property should return a set of the following promises:
```

Comment:
I would indicate specifically who has to do what, i.e. "to use this feature, you must correctly implement and provide a set of following functions":